File: semester2/5_Fuzzy-Software-Technologies/aics-fuzzy-software-technologies/ticket-backend/src/main/java/aics/domain/fuzzy/dtos/FuzzySearchFiltersDto.java
package aics.domain.fuzzy.dtos;

import aics.domain.fuzzy.constants.FuzzySearchChoices;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.RequiredArgsConstructor;
import lombok.experimental.Accessors;

import java.io.Serializable;

@Data
@Accessors(chain = true)
@RequiredArgsConstructor
@AllArgsConstructor
public class FuzzySearchFiltersDto implements Serializable {
    private FuzzySearchChoices choice1;
    private FuzzySearchChoices choice2;
    private FuzzySearchChoices choice3;
    private FuzzySearchChoices choice4;
    private boolean yearCostCriteria;
    private boolean durationCostCriteria;
}

File: semester2/5_Fuzzy-Software-Technologies/aics-fuzzy-software-technologies/ticket-frontend/src/modules/fuzzy/dtos/fuzzy-search-filters-dto.ts
import { FuzzySearchChoices } from '../fuzzy-constants';

export class FuzzySearchFiltersDto {
    choice1: FuzzySearchChoices;
    choice2: FuzzySearchChoices;
    choice3: FuzzySearchChoices;
    choice4: FuzzySearchChoices;
    yearCostCriteria: boolean;
    durationCostCriteria: boolean;

    constructor(obj: {
        choice1: FuzzySearchChoices,
        choice2: FuzzySearchChoices,
        choice3: FuzzySearchChoices,
        choice4: FuzzySearchChoices,
        yearCostCriteria: boolean,
        durationCostCriteria: boolean,
    }) {
        this.choice1 = obj.choice1;
        this.choice2 = obj.choice2;
        this.choice3 = obj.choice3;
        this.choice4 = obj.choice4;
        this.yearCostCriteria = obj.yearCostCriteria;
        this.durationCostCriteria = obj.durationCostCriteria;
    }

    static fromObj(obj: any): FuzzySearchFiltersDto {
        if (!obj) {
            throw new Error('obj cannot be null');
        }
        return new FuzzySearchFiltersDto({
            choice1: obj.choice1,
            choice2: obj.choice2,
            choice3: obj.choice3,
            choice4: obj.choice4,
            yearCostCriteria: obj.yearCostCriteria,
            durationCostCriteria: obj.durationCostCriteria,
        });
    }

    deepClone(): FuzzySearchFiltersDto {
        return new FuzzySearchFiltersDto({
            choice1: this.choice1,
            choice2: this.choice2,
            choice3: this.choice3,
            choice4: this.choice4,
            yearCostCriteria: this.yearCostCriteria,
            durationCostCriteria: this.durationCostCriteria,
        })
    }
}